src/openms/source/METADATA/ID/IdentificationDataConverter.cpp

```diff
@@ -0,0 +1,888 @@
+        match.scores.push_back(make_pair(score_ref, hit.getScore()));
+        id_data.registerMoleculeQueryMatch(match);
+      }
+    }
+    progresslogger.endProgress();
+  }
+
+
+  void IdentificationDataConverter::exportIDs(
+    const IdentificationData& id_data, vector<ProteinIdentification>& proteins,
+    vector<PeptideIdentification>& peptides, const String& protein_score,
+    const String& peptide_score, bool export_oligonucleotides)
+  {
+    // @TODO: export ParentMoleculeGroups
+
+    proteins.clear();
+    peptides.clear();
+
+    // "DataQuery" roughly corresponds to "PeptideIdentification",
+    // "DataProcessingStep" roughly corresponds to "ProteinIdentification":
+    map<pair<IdentificationData::DataQueryRef,
+             IdentificationData::ProcessingStepRef>,
+        pair<vector<PeptideHit>, IdentificationData::ScoreTypeRef>> psm_data;
+    // we only export peptides and proteins (or oligos and RNAs), so start by
+    // getting the PSMs (or OSMs):
+    const String& ppm_error_name = Constants::PRECURSOR_ERROR_PPM_USERPARAM;
+    for (const IdentificationData::MoleculeQueryMatch& query_match :
+           id_data.getMoleculeQueryMatches())
+    {
+      PeptideHit hit;
+      const IdentificationData::ParentMatches* parent_matches_ptr;
+      if (!export_oligonucleotides) // export peptides
+      {
+        if (query_match.getMoleculeType() !=
+            IdentificationData::MoleculeType::PROTEIN) continue;
+        static_cast<MetaInfoInterface&>(hit) = query_match;
+        IdentificationData::IdentifiedPeptideRef peptide_ref =
+          query_match.getIdentifiedPeptideRef();
+        hit.setSequence(peptide_ref->sequence);
+        parent_matches_ptr = &(peptide_ref->parent_matches);
+      }
+      else
+      {
+        if (query_match.getMoleculeType() !=
+            IdentificationData::MoleculeType::RNA) continue;
+        static_cast<MetaInfoInterface&>(hit) = query_match;
+        IdentificationData::IdentifiedOligoRef oligo_ref =
+          query_match.getIdentifiedOligoRef();
+        hit.setMetaValue("label", oligo_ref->sequence.toString());
+        parent_matches_ptr = &(oligo_ref->parent_matches);
+      }
+      hit.setCharge(query_match.charge);
+      if (query_match.metaValueExists(ppm_error_name))
+      {
+        hit.setMetaValue(ppm_error_name,
+                         query_match.getMetaValue(ppm_error_name));
+      }
+      for (const auto& pair : *parent_matches_ptr)
+      {
+        IdentificationData::ParentMoleculeRef parent_ref = pair.first;
+        for (const IdentificationData::MoleculeParentMatch& parent_match :
+               pair.second)
+        {
+          PeptideEvidence evidence;
+          evidence.setProteinAccession(parent_ref->accession);
+          evidence.setStart(parent_match.start_pos);
+          evidence.setEnd(parent_match.end_pos);
+          evidence.setAABefore(parent_match.left_neighbor[0]);
+          evidence.setAAAfter(parent_match.right_neighbor[0]);
```

Comment:
what happens if the string is empty? sounds like a nasty segfault is possible here!